scripts/pylib/twister/twisterlib.py

```diff
@@ -2767,6 +2770,43 @@ def process(self, pipeline, done, message, lock, results):
             else:
                 self.cleanup_artifacts()
 
+    def amend_test_instance(self, results):
+        lst_file = os.path.join(self.build_dir, "zephyr", "zephyr.lst")
+        if os.path.isfile(lst_file):
+            logger.debug(f"zephyr.lst located: {lst_file}")
+        else:
+            # no zephyr.lst file, cannot do symbol-based amending
+            logger.debug(f"zephyr.lst NOT located: {lst_file}")
+            return
+
+        yaml_testsuite_name = self.instance.testsuite.id
+        logger.debug(f"amending yaml test suite: {yaml_testsuite_name}")
+
+        with open(lst_file, 'r') as fp:
+            lst_content = fp.read()
+            logger.debug(f"test instance {self.instance.name} has {len(self.instance.testcases)} cases.")
+            for testcase in self.instance.testcases:
+                # identifier is "yaml_testsuite.test_function" or just "yaml_testsuite"
+                identifier = testcase.name
+                if identifier == yaml_testsuite_name:
+                    # Sometimes twsiter add yaml test suite name to a test instance as a test case.
+                    # Such test case has no real test function.
+                    # Skip it to make stats right
+                    continue
+
+                test_func_name1 = f"test_{identifier.split('.')[-1]}"
+                test_func_name2 = f"{identifier.split('.')[-1]}"
+                # If the .lst file exists and the test case function not in it,
+                # it should be an un-intended test function for this test instance build.
+                # If this is an old test plan and contains test cases collected at runtime
+                # by checking the log, such test cases may not have functions in .lst file.
+                # Also remove them here, which should be OK because they will still be added back at runtime.
```

Comment:
I believe these comments are not accurate anymore, right? Can you fix them?

Reply:
Thanks. It is fixed now.